lib/opal/builder.rb

```diff
@@ -153,25 +150,24 @@ def tree_requires(asset, path)
     end
 
     def processor_for(source, filename, path, options)
-      processor = processors.find { |p| p.match? path } or
-        raise ProcessorNotFound, "can't find processor for filename: #{filename.inspect}, path: #{path.inspect}, source: #{source.inspect}, processors: #{processors.inspect}"
+      (processor = processors.find { |p| p.match? path }) ||
+        raise(ProcessorNotFound, "can't find processor for filename: #{filename.inspect}, path: #{path.inspect}, source: #{source.inspect}, processors: #{processors.inspect}")
```

Comment:
Yes, parentheses around assignment are not required here.
About `&&/||` and `and/or` - rubocop explicitly [says](https://github.com/bbatsov/ruby-style-guide#no-and-or-or) that `The and and or keywords are banned`.

`or` suites better for this case because it's allowed to write `expr or raise msg`, but `expr || raise(msg)` also looks fine for me.

Reply:
I'm always confused with `and/or`.
:+1: for banning them.

Reply:
> rubocop explicitly says that The and and or keywords are banned.

That's their prerogative, but any choice is far from untouchable, of course we're in compromise-land here and I'd prefer not to ban anything